corehq/apps/users/views/mobile/users.py

```diff
@@ -730,19 +733,15 @@ def check_username(self, in_data):
             return HttpResponseBadRequest('You must specify a username')
         if username == 'admin' or username == 'demo_user' or username == ANONYMOUS_USERNAME:
             return {'error': _('Username {} is reserved.').format(username)}
-        if '@' in username:
-            return {
-                'error': _('Username {} cannot contain "@".').format(username)
-            }
-        if '&' in username:
-            return {
-                'error': _('Username {} cannot contain "&".').format(username)
-            }
-        if ' ' in username:
+        try:
+            validate_email("{}@example.com".format(username))
+            if BAD_MOBILE_USERNAME_REGEX.search(username) is not None:
+                raise ValidationError("Username contained an invalid character")
+        except ValidationError:
             return {
-                'error': _('Username {} cannot contain '
-                           'spaces.').format(username)
+                'error': _("Username may only contain letters, numbers, + or .")
```

Comment:
This is kind of a lie. there are a few other characters that are allowed, but I figured keeping this short and simple is better. ($ is allowed because I've created a some amount of mobile workers with name j$)

Reply:
I concur that keeping it simple is better, might be better to make it even vaguer and say "Username may not contain special characters." If you keep the current message, I'd add `_` and maybe `-`.